test/PlayerActions/PAY_FINE.test.js

```diff
@@ -0,0 +1,121 @@
+const expect = require("chai").expect;
+const EventEmitter = require("events");
+const sinon = require("sinon");
+
+const mockUIFactory = require('../mocks/UI');
+
+const { GameState } = require("../../entities/GameState");
+const { createPlayer } = require("../testutils");
+
+describe("PlayerActions -> PAY_FINE", () => {
+  let gameState;
+  let mockUI;
+  let eventBusEmitter;
+  let PLAYER_ACTIONS;
+  
+  beforeEach(() => {
+    gameState = new GameState();
+    eventBusEmitter = new EventEmitter();
+    mockUI = mockUIFactory();
+    
+    gameState.players = [createPlayer({ name: "player1" })];
+    PLAYER_ACTIONS = require("../../entities/PlayerActions")(eventBusEmitter, mockUI, gameState);
+  });
+  
+  afterEach(() => {
+    // Restore the default sandbox here
+    sinon.restore();
+  });
+
+  describe("payFine", () => {
+    // The method under test
+    let methodUnderTest;
+    // Output Event from "rollDice"'s emit
+    const bankruptcyEvent = "BANKRUPTCY";
+    const liquidationEvent = "LIQUIDATION";
+    let bankruptcySpy, liquidationSpy;
+
+    beforeEach(() => {
+      bankruptcySpy = sinon.spy();
+      liquidationSpy = sinon.spy();
+      // Clear default listeners to prevent side-effects*
+      eventBusEmitter.removeAllListeners(bankruptcyEvent);
+      eventBusEmitter.on(bankruptcyEvent, bankruptcySpy);
```

Comment:
We need to clear the listeners because we don't recreate the event emitter every time?

Reply:
We do recreate them, actually (line 18) but I am clearing them to prevent duplicate methods from being called;
e.g., if I am testing for bankruptcy and don't disable the Bankruptcy event, during my test it might end the player's turn forcefully and go to the next player's turn, which would change the game state and wreck my tests

Reply:
If you remove the `removeAllListeners`, do your tests actually fail? Shouldn't they be added new each time because `beforeEach` runs?

Reply:
Yeah, we cannnnn